cocos2d/core/CCGame.js

```diff
@@ -356,31 +360,51 @@ var game = /** @lends cc.game# */{
                 }
                 else if ( !(node.parent instanceof cc.Scene) ) {
                     cc.warn('The node can not be made persist because it\'s not under root node.');
-                    return;
+                    return -1;
                 }
                 else if (node.parent !== scene) {
                     cc.warn('The node can not be made persist because it\'s not in current scene.');
-                    return;
+                    return -1;
                 }
                 this._persistRootNodes[id] = node;
                 node._persistNode = true;
+                return id;
             }
         }
     },
+    getPersistRootNode:function(key){
+         if (!key) {
+            cc.warn(' getPersistRootNode key==null');
```

Comment:
Make sure any runtime log information in cc.log , cc.error, cc.warn or cc.assert has been moved into DebugInfos.js with an ID